packages/select-rich/src/LionSelectRich.js

```diff
@@ -132,16 +107,40 @@ export class LionSelectRich extends OverlayMixin(
     return this._listboxNode.querySelector(`#${this._listboxActiveDescendant}`);
   }
 
-  get checkedIndex() {
-    if (this.modelValue) {
-      return this.modelValue.findIndex(el => el.value === this.checkedValue);
+  get modelValue() {
+    const el = Array.from(this._listboxNode.children).find(option => option.checked);
+    return el ? el.modelValue.value : '';
+  }
+
+  set modelValue(value) {
+    const el = Array.from(this._listboxNode.children).find(
+      option => option.modelValue.value === value,
+    );
+
+    if (el) {
+      el.checked = true;
+    } else {
+      // cache user set modelValue until firstUpdated, and then try it again
+      this.__cachedUserSetModelValue = value;
     }
-    return -1;
+
+    this.__onModelValueChanged();
+    this.requestUpdate('modelValue');
+  }
+
+  get checkedIndex() {
+    let checkedIndex = -1;
+    Array.from(this._listboxNode.children).forEach((option, i) => {
```

Comment:
Note, using listboxNode directly , because formElements array is not great with timing and keeping the right order of the elements... it only appends to the end of the array even if you prepend a new <lion-option> node to the listbox. Caused quite some bugs / weird behavior. See advanced use case tests for a bit more context